src/my_std.ml

```diff
@@ -279,7 +279,7 @@ let sys_command =
   match Sys.os_type with
   | "Win32" -> fun cmd ->
       if cmd = "" then 0 else
-      let cmd = "bash --norc -c " ^ Filename.quote cmd in
+      let cmd = "bash --norc -c " ^ Filename.quote cmd ^ " 2>&1" in
```

Comment:
Can you explain why we need this change?

Reply:
I'm not happy with this change but it makes the testsuite pass.

I think the explanation is as follow:
- if `is_degraded` or Sys`.win32` is true, `execute_many` falls back to `sys_command` to implement `execute_many`.
- otherwise, the logic from `Ocamlbuild_executor.execute` is used, which reads both stderr and stdout into a `job_buffer`.
- The testuite sometime expect some output on stdout.  

I'm still looking into a better solution.

Reply:
I've moved the stderr redirection to `command.ml` where I think it belongs and added a small comment.

Reply:
I went to look at the code of ocamlbuild_executor to understand what is going on. `ocamlbuild_executor` does not just redirect stderr into stdout, it lets the command write to different places *and then* on job termination it adds stdout, then stderr, to the job buffer, in that order. The change that you implemented does not match this behavior, it only works for commands that have only stdout or only stderr output.

Reply:
Out of curiosity: in the `execute_many` definition, have you tried removing the "or we are on Windows" condition, to get the ocamlbuild_executor logic to run also on Windows? When ocamlbuild was first written, I think the Windows support in stdlib and Unix was weaker than it is now, there is a small chance that removing those alternate paths could work today.

(We could probably also get rid of the "is_degraded" notion, which I think was designed to let ocamlbuild build the compiler in constrained environments. ocamlbuild is not considered as build system for the OCaml compiler anymore, so maybe we could forget about this case altogether. But this is more work than I was considering putting in ocamlbuild again.)

Reply:
> I went to look at the code of ocamlbuild_executor to understand what is going on. `ocamlbuild_executor` does not just redirect stderr into stdout, it lets the command write to different places _and then_ on job termination it adds stdout, then stderr, to the job buffer, in that order. The change that you implemented does not match this behavior, it only works for commands that have only stdout or only stderr output.

Looking at the code quickly, my impression is that it can interleave stdout and stderr. In the main `loop`, you can see
```
chrfds, do_read ~loop:false;
```
which tries to read any both stdout and stderr and push the content to `job_buffer`.

Maybe I'm misunderstanding the logic.

Reply:
Ok, this suggests that I misunderstood, and there may be no good way to ensure that we get the same output. In that context I am okay with your current choice even though it is sort of a hack.

Reply:
> Out of curiosity: in the `execute_many` definition, have you tried removing the "or we are on Windows" condition, to get the ocamlbuild_executor logic to run also on Windows? When ocamlbuild was first written, I think the Windows support in stdlib and Unix was weaker than it is now, there is a small chance that removing those alternate paths could work today.
> 
> (We could probably also get rid of the "is_degraded" notion, which I think was designed to let ocamlbuild build the compiler in constrained environments. ocamlbuild is not considered as build system for the OCaml compiler anymore, so maybe we could forget about this case altogether. But this is more work than I was considering putting in ocamlbuild again.)

Done in #333 and #332